File: android/tutorials/tutorial2-composing-ribs-finish-code/src/test/java/com/uber/rib/root/RootInteractorTest.java
package com.uber.rib.root;

import com.uber.rib.core.InteractorHelper;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import io.reactivex.Observable;

import static org.mockito.Matchers.any;
import static org.mockito.Mockito.never;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class RootInteractorTest {

  @Mock RootInteractor.RootPresenter presenter;
  @Mock RootRouter router;

  private RootInteractor interactor;

  @Before
  public void setup() {
    MockitoAnnotations.initMocks(this);

    interactor = TestRootInteractor.create(presenter);
  }

}

File: android/demos/compose/src/test/kotlin/com/uber/rib/compose/RootInteractorTest.kt
package com.uber.rib.compose

import com.uber.rib.compose.root.main.MainInteractor
import com.uber.rib.compose.root.main.MainRouter
import com.uber.rib.core.RibTestBasePlaceholder
import com.uber.rib.core.InteractorHelper

import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.MockitoAnnotations

class RootInteractorTest : RibTestBasePlaceholder() {

  @Mock internal lateinit var presenter: MainInteractor.RootPresenter
  @Mock internal lateinit var router: MainRouter

  private var interactor: MainInteractor? = null

  @Before
  fun setup() {
    MockitoAnnotations.initMocks(this)

    interactor = TestRootInteractor.create(presenter)
  }

  /**
   * TODO: Delete this example and add real tests.
   */
  @Test
  fun anExampleTest_withSomeConditions_shouldPass() {
    // Use InteractorHelper to drive your interactor's lifecycle.
    InteractorHelper.attach<MainInteractor.RootPresenter, MainRouter>(interactor!!, presenter, router, null)
    InteractorHelper.detach(interactor!!)

    throw RuntimeException("Remove this test and add real tests.")
  }
}